test/task/ReceiveRemoteVideoPauseResumeTask.test.ts

```diff
@@ -286,5 +286,168 @@ describe('ReceiveRemoteVideoPauseResumeTask', () => {
+        .be.false;
+      expect(context.videoTileController.getVideoTileForAttendeeId('attendee-8')).to.not.be
+        .undefined;
+      expect(context.videoTileController.getVideoTileForAttendeeId('attendee-8').state().paused).to
+        .be.false;
+      expect(context.videoTileController.getVideoTileForAttendeeId('attendee-9')).to.not.be
+        .undefined;
+      expect(context.videoTileController.getVideoTileForAttendeeId('attendee-9').state().paused).to
+        .be.false;
+    });
+
+    it('Will not resume user paused tiles', async () => {
+      context.videoTileController = new DefaultVideoTileController(
+        new DefaultVideoTileFactory(),
+        context.audioVideoController,
+        null
+      );
+
+      const policy = new VideoAdaptiveProbePolicy(new NoOpDebugLogger());
+      class MockObserver implements VideoDownlinkObserver {
+        tileWillBePausedByDownlinkPolicy = sinon.stub();
+        tileWillBeUnpausedByDownlinkPolicy = sinon.stub();
+      }
+      const observer = new MockObserver();
+      policy.addObserver(observer);
+      context.videoDownlinkBandwidthPolicy = policy;
+
+      const pausedResumeFrame = SdkPauseResumeFrame.create();
+      pausedResumeFrame.groupIds = [7, 8];
+      pausedResumeFrame.streamIds = new DefaultVideoStreamIdSet([7, 8]).array();
+      const pauseFrame = SdkSignalFrame.create();
+      pauseFrame.type = SdkSignalFrame.Type.PAUSE;
+      pauseFrame.pause = pausedResumeFrame;
+      const pauseEvent = new SignalingClientEvent(
+        signalingClient,
+        SignalingClientEventType.ReceivedSignalFrame,
+        pauseFrame
+      );
+      await task.run();
+
+      const resumeFrame = SdkSignalFrame.create();
+      resumeFrame.type = SdkSignalFrame.Type.RESUME;
+      const frame2 = SdkPauseResumeFrame.create();
+      frame2.groupIds = [7, 8];
+      frame2.streamIds = new DefaultVideoStreamIdSet([7, 8]).array();
+      resumeFrame.pause = frame2;
+      const resumeEvent = new SignalingClientEvent(
+        signalingClient,
+        SignalingClientEventType.ReceivedSignalFrame,
+        resumeFrame
+      );
+
+      context.videoStreamIndex = prepareIndex([7, 8]);
+      task.updateSubscribedGroupdIds(new Set([7, 8]));
+      task.handleSignalingClientEvent(pauseEvent);
+
+      expect(context.videoTileController.getVideoTileForAttendeeId('attendee-7')).to.not.be
+        .undefined;
+      expect(context.videoTileController.getVideoTileForAttendeeId('attendee-7').state().paused).to
+        .be.true;
+      expect(context.videoTileController.getVideoTileForAttendeeId('attendee-8')).to.not.be
+        .undefined;
+      expect(context.videoTileController.getVideoTileForAttendeeId('attendee-8').state().paused).to
+        .be.true;
+
+      // Additional test to allow reseting paused groups
+      task.updateSubscribedGroupdIds(new Set([]));
+
+      // Manually pause
+      context.videoTileController.getVideoTileForAttendeeId('attendee-7').stateRef().paused = true;
```

Comment:
Should we use actual API used to pause if paused by the builder application? It may happen so that approach is not correctly pausing the video tile and help us catch it.

Reply:
sure, i forgot that API existed.